en/dev_log/log_encryption.md

```diff
@@ -0,0 +1,179 @@
+# Log Encryption
+
+The [System Logger](../modules/modules_system.md#logger) can be used to create encrypted logs, which may then be decrypted manually before analysis.
+
+The encryption algorithm used is XChaCha20, and the default wrapping algorithm used is RSA2048-OAEP.
+
+::: warning
+Log encryption is not enabled by default in PX4 firmware builds.
+To use it you will need to build firmware with this feature enabled and then upload it to the flight controller (see instructions below).
+:::
+
+::: info
+The encryption algorithm used can be changed by modifying [SDLOG_ALGORITHM](../advanced_config/parameter_reference.md#SDLOG_ALGORITHM).
+However at time of writing, only XChaCha20 can actually be used (AES can be selected, but there is no implementation).
+:::
+
+## How ULog Encryption Works
+
+When the ULog file is first opened:
+
+- If [SDLOG_ALGORITHM](../advanced_config/parameter_reference.md#SDLOG_ALGORITHM) is set to default, a symmetric XChaCha20 key is generated and wrapped (encrypted) with RSA2048 public key encryption
+- This wrapped key is stored on the SD Card with the ".ulgk" suffix.
+
```

Comment:
I'd quite like to make this generic of XChaCha20 if possible. 

I understand that the system generates a symmetric key that is encrypted using the owners public key and stored on the SDK card. The symmetric key is used to encrypt the log. Only people who have the private key can decrypt the symmetric key file to get the key, and then use it to decrypt the log.

1. Is the symmetric key generated each time a new log is it created to go with the log, or is/can the same key used for multiple logs? I assume a new key each time!
2. If you used AES (were it supported) would the same process apply, but with the AES symmetric key? I.e. an AES key would be encrypted etc. Would the file names be the same in this case - .e. ".ulgk"?

Reply:
> Is the symmetric key generated each time a new log is it created to go with the log, or is/can the same key used for multiple logs? I assume a new key each time!

A new key is generated for each new log. Basically, each log generates two files, the encrypted log (.ulgc), and the associated symmetric key (.ulgk).

> If you used AES (were it supported) would the same process apply, but with the AES symmetric key? I.e. an AES key would be encrypted etc. Would the file names be the same in this case - .e. ".ulgk"?

I am not well versed enough into encryption algorithms to know the answer to this question. Since they are both symmetric key algorithms, I would assume the implementation would be similar, but there is likely nuance to it that I am missing.

> How is the symmetric key protected on the device? I..e. is it amenable to being found by someone (other than the encrypted versions), can it be replaced somehow so that a "bad" key is always used?

The symmetric key itself is encrypted, so you would need to have access to the private rsa key to unwrap the symmetric key to decrypt the log.

Reply:
I have rewritten this as though it is generic, prefixed with a note that it is _likely_ to be the same for other algorithms